lib/spack/spack/cmd/__init__.py

```diff
@@ -60,11 +59,19 @@ def all_commands():
     global _all_commands
     if _all_commands is None:
         _all_commands = []
-        for file in os.listdir(spack.paths.command_path):
-            if file.endswith(".py") and not re.search(ignore_files, file):
-                cmd = re.sub(r'.py$', '', file)
-                _all_commands.append(cmd_name(cmd))
+        extensions = spack.config.get('config:extensions') or []
```

Comment:
I think the `extensions` module should have a method like `get_command_paths` and should handle retrieving `config:extensions` in that method (so basically, make a function that combines this line and `spack.extensions.command_paths...` into one call), like `get_template_dirs` in `tengine`.

Reply:
Done in 2de1949